modules/geoedgeRtdProvider.js

```diff
@@ -0,0 +1,238 @@
+    '%_isHb!': true,
+    '%_hbcid!': bid.creativeId || '',
+    '%%PATTERN:hb_pb%%': bid.pbHg,
+    '%%SITE%%': location.hostname,
+    '%_pimp%': PV_ID
+  };
+}
+
+/**
+ * replace macro placeholders in a string with values from a dictionary
+ * @param {string} wrapper
+ * @param {Object} macros
+ * @return {string}
+ */
+function replaceMacros(wrapper, macros) {
+  let key;
+  for (key in macros) {
+    wrapper = wrapper.replace(key, replacer(macros[key]));
+  }
+  return wrapper;
+}
+
+/**
+ * build final creative html with creative wrapper
+ * @param {Object} bid
+ * @param {string} wrapper
+ * @param {string} html
+ * @return {string}
+ */
+function buildHtml(bid, wrapper, html, key) {
+  let macros = getMacros(bid, key);
+  wrapper = replaceMacros(wrapper, macros);
+  return wrapHtml(wrapper, html);
+}
+
+/**
+ * muatates the bid ad property
+ * @param {Object} bid
+ * @param {string} ad
+ */
+function mutateBid(bid, ad) {
+  bid.ad = ad;
+}
+
+/**
+ * wraps a bid object with the creative wrapper
+ * @param {Object} bid
+ * @param {string} key
+ */
+export function wrapBidResponse(bid, key = providerParams.key) {
+  let wrapped = buildHtml(bid, wrapper, bid.ad, key);
+  mutateBid(bid, wrapped);
+}
+
+/**
+ * checks if bidder's bids should be monitored
+ * @param {string} bidder
+ * @return {boolean}
+ */
+function isSupportedBidder(bidder) {
+  return isEmpty(providerParams.bidders) || providerParams.bidders[bidder] === true;
+}
+
+/**
+ * checks if bid should be monitored
+ * @param {Object} bid
+ * @return {boolean}
+ */
+function shouldWrap(bid) {
+  let supportedBidder = isSupportedBidder(bid.bidderCode);
+  let wap = providerParams.wap ? preloaded : true;
+  return wrapperReady && supportedBidder && wap;
+}
+
+function conditionallyWrap(bid) {
+  if (shouldWrap(bid)) {
+    wrapBidResponse(bid);
+  }
+}
+
+function init(config, gdpr, usp) {
+  return true;
+}
+
+/** @type {RtdSubmodule} */
+export const geoedgeSubmodule = {
+  /**
+     * used to link submodule with realTimeData
+     * @type {string}
+     */
+  name: SUBMODULE_NAME,
+  /**
+     * get data and send back to realTimeData module
+     * @function
+     * @param {adUnit[]} adUnits
+     * @param {function} onDone
+     */
+  init,
+  onBidResponseEvent: conditionallyWrap
+};
+
+/**
+ * sets the module params from config
+ * @param {Object} realTimeData
+ */
+export function setParams(realTimeData) {
+  let dataProviders = realTimeData && realTimeData.dataProviders;
+  let geoedge;
+  try {
+    geoedge = dataProviders && find(dataProviders, provider => provider.name && provider.name.toLowerCase() === SUBMODULE_NAME);
+  } catch (e) { }
+  let params = geoedge && geoedge.params;
+  if (!params || !params.key) {
+    logError('missing key for geoedge RTD module provider');
+  } else {
+    providerParams = params;
+    fetchWrapper(setWrapper);
+    preloadClient(params.key);
+  }
+}
+
+let unsubscribe = config.getConfig(MODULE_NAME, ({ realTimeData }) => {
```

Comment:
You should not be listening to getConfig here. As described in https://docs.prebid.org/dev-docs/add-rtd-submodule.html, the conditionallyWrap() function will be called with these params: bidResponse, config, userConsent. You will get your params from there instead.